ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -3821,6 +3818,28 @@ void HqlCppTranslator::convertLogicalToActivities(WorkflowItem & curWorkflow)
 }
 
 //------------------------------------------------------------------------
+static HqlTransformerInfo setResultToExtractTransformerInfo("SetResultToExtractTransformer");
+SetResultToExtractTransformer::SetResultToExtractTransformer()
+: NewHqlTransformer(setResultToExtractTransformerInfo)
+{
+}
+
+
+IHqlExpression * SetResultToExtractTransformer::createTransformed(IHqlExpression * expr)
+{
+    OwnedHqlExpr transformed = PARENT::createTransformed(expr);
+    if (transformed->getOperator() == no_setresult)
+    {
+        OwnedHqlExpr normalized = convertSetResultToExtract(transformed);
+        if (normalized && (normalized != transformed))
+            transformed.setown(transform(normalized));
```

Comment:
minor: I think this can say

if (normalized)
   transformed.set(normalized);

There isn't any scope for that to be retransformed - unlike some of the items in the previous context.